### PR TITLE
Various improvements after CSS merge

### PR DESCRIPTION
**Pull request by vilchik-elena**
* Update README
* Update documentation.md (merge JS and CSS, TS 4.4)
* create separate interfaces for AnalysisRequest for JS and CSS
* drop mock-server.test.ts and move its tests into `analyzer.test.ts`
* Rename `configFile` into `stylelintConfig` in AnalysisRequest
* Move CssRuleSensor to `eslint` package
* Fix imports formatting 
* Other quality issues

### CODE REVIEW

File: eslint-bridge/tests/analyzer.test.ts [hunk 1]
```diff
@@ -69,6 +70,7 @@ describe('#analyzeJavaScript', () => {
       filePath,
       fileContent: codeToTest,
       fileType: 'MAIN',
+      tsConfigs: [],
```
Comment (vilchik-elena) [inline eslint-bridge/tests/analyzer.test.ts:73]: to reproduce production analysis we should provide tsconfig here.

File: sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/css/metrics/CssMetricSensor.java [hunk 2]
```diff
@@ -37,13 +37,13 @@
 import org.sonar.api.utils.log.Loggers;
 import org.sonar.plugins.javascript.css.CssLanguage;
 
-public class MetricSensor implements Sensor {
+public class CssMetricSensor implements Sensor {
```
Comment (yassin-kammoun-sonarsource) [inline sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/css/metrics/CssMetricSensor.java:40]: I have a vague memory of hardcoded sensor names somewhere for speed of analysis. Will this change impact it ?

**Review by yassin-kammoun-sonarsource — APPROVED** [review]
LGTM. I suggested minor changes that you are free to ignore if irrelevant. Also, we need to create tickets for the pieces of code (message trimming, error parsing, metric) that we expect to move from CSS sensor to the bridge.

File: sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/css/metrics/CssMetricSensor.java (hunk 2 above)
Comment (vilchik-elena) [reply sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/css/metrics/CssMetricSensor.java:40]: Nope, we are relying there on sensor's name, set in `describe` method

File: sonar-javascript-plugin/src/main/resources/static/documentation.md [hunk 3]
```diff
@@ -1,39 +1,42 @@
 ---
-title: JS / TS / CSS
+title: JavaScript / TypeScript / CSS
```
[the remark this answers is not in the document]
Comment (vilchik-elena) [reply sonar-javascript-plugin/src/main/resources/static/documentation.md:2]: I will remove spaces but prefer to keep CSS last to keep JavaScript more visible

**Pull request merged**